comms/discovery/db/queries/get_chat_blocked_users.go

```diff
@@ -35,3 +36,46 @@ func CountChatBlocks(q db.Queryable, ctx context.Context, arg CountChatBlocksPar
 	err := q.GetContext(ctx, &count, countChatBlocks, arg.User1, arg.User2)
 	return count, err
 }
+
+const bulkGetChatBlocksOrReceivedBlocks = `
+select blocker_user_id, blockee_user_id from chat_blocked_users where (blocker_user_id = :SenderUserID and blockee_user_id in (:ReceiverUserIDs)) or (blockee_user_id = :SenderUserID and blocker_user_id in (:ReceiverUserIDs))
+`
+
+type BulkGetChatBlocksOrReceivedBlocksParams struct {
+	SenderUserID    int32   `json:"sender_user_id"`
+	ReceiverUserIDs []int32 `json:"receiver_user_ids"`
+}
+
+type BulkGetChatBlocksOrReceivedBlocksRow struct {
+	BlockerUserID int32 `db:"blocker_user_id" json:"blocker_user_id"`
+	BlockeeUserID int32 `db:"blockee_user_id" json:"blockee_user_id"`
+}
+
+func BulkGetChatBlocksOrReceivedBlocks(q db.Queryable, ctx context.Context, arg BulkGetChatBlocksOrReceivedBlocksParams) ([]BulkGetChatBlocksOrReceivedBlocksRow, error) {
```

Comment:
nit: this is kind of a mouthful 😅 also kinda worried about the "block" overloading with like chain blocks

Maybe `BulkGetChatBlockedOrBlocking`?

"blocked" = they blocked me
"blocking" = I blocked them

or could keep blocker/blockee naming at this level: `BulkGetChatBlockersOrBlockees`